woocommerce/woocommerce-ios
# Dynamic Dashboard: Stats and chart views for store performance card

Part of #12465 


## Description

This PR adds stats and chart views for the new store performance card. I reuse `StoreStatsPeriodViewModel` to retain most of the existing logic, and copied the checks for the custom range redacted view from the legacy view controllers.

Changes:
- Updated `DashboardView` and its hosting controller to handle navigation to the analytic hub and presenting the custom range alert from the performance card.
- Updated the performance card to:
  - Display the selected date for custom time ranges. We don't have design for how to display this, so I'm replacing the date range with the selected date. I tried displaying the date below the range but it looks odd - please feel free to suggest a different option if you have an idea.
  - Add a view for stat values and redacted views.
  - Add chart view using the new chart from #12467.
  - Add a button to navigate to analytics hub.
  - Track interactions with the chart and time range picker.
- Updated `StoreStatsChart` to:
  - disable selecting chart when there is no revenue. 
  - support deselect points on the chart.
  - update no revenue view color.
 
Due to the large number of changes, unit tests will be added in a separate PR.

## Testing instructions

- Enable the feature flag `dynamicDashboard` and build the app.
- Ensure that store performance card is enabled on the dashboard screen.
- Confirm that the card works properly:
  - Switch different time ranges and confirm that the stats and chart are updated correctly.
  - Tap on the chart, confirm that tapping only works when there is any revenue. Confirm that the selected date and stats values are displayed upon tapping. Tapping the same point again should deselect it and the values and charts should be back to the unselected state.
  - Switch between different stores, confirm that the visitor and conversion stats are redacted for JCP sites and when authenticated without WPCom.
  - Switch to a WPCom store and select a custom range. Confirm that the visitor and conversion stats are redacted for ranges longer than 1 day but not for 1 day. Select any point on the chart, confirm that the visitor stats are not redacted for ranges longer than 1 day.
  - Tap on the redacted view for custom range, confirm that the alert explaining the reason is displayed.
  - Tap on View all store analytics, confirm that the app navigates to analytics hub.
  - Confirm that tracking events are sent properly when interacting with the chart and time range picker.

## Screenshots


No revenue | With revenue
--- | ---
<img src="https://github.com/woocommerce/woocommerce-ios/assets/5533851/ea4e82fc-fa0d-49d5-9011-9788671a88f5" width=320 /> | <img src="https://github.com/woocommerce/woocommerce-ios/assets/5533851/2dbd29ec-0c3e-4065-91b8-6b8030122de8" width=320 />


---
